src/data/list/basic.lean

```diff
@@ -4552,6 +4552,33 @@ theorem prod_range_succ {α : Type u} [monoid α] (f : ℕ → α) (n : ℕ) :
 by rw [range_concat, map_append, map_singleton,
   prod_append, prod_cons, prod_nil, mul_one]
 
+@[to_additive]
+theorem prod_range_succ' {α : Type u} [monoid α] (f : ℕ → α) (n : ℕ) :
```

Comment:
I can't think of a better name, but perhaps we can add a doc-string saying what it does, and how it's related to the unprimed version.

(Perhaps, in fact, in the long run we should add the doc-string linter requirement to all theorems/lemmas ending in `'`.)

Reply:
It agrees with `finset` version in `algebra/big_operators`